integration/krew/krew.go

```diff
@@ -0,0 +1,118 @@
+package krew
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/krew/pkg/testutil"
+)
+
+// KrewTest is used to set up `krew` integration tests.
+type KrewTest struct {
+	t       *testing.T
+	args    []string
+	env     []string
+	tempDir *testutil.TempDir
+}
+
+// NewKrewTest creates a fluent krew KrewTest.
+func NewKrewTest(t *testing.T) (*KrewTest, func()) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	return &KrewTest{
+		t:       t,
+		env:     []string{fmt.Sprintf("KREW_ROOT=%s", tempDir.Root())},
+		tempDir: tempDir,
+	}, cleanup
+}
+
+// Cmd sets the arguments to krew.
+func (k *KrewTest) Cmd(args ...string) *KrewTest {
+	k.args = args
+	return k
+}
+
+// Root returns the krew root directory for this test.
+func (k *KrewTest) Root() string {
+	return k.tempDir.Root()
+}
+
+// WithIndex initializes the index with the actual krew-index from github/kubernetes-sigs/krew-index.
+func (k *KrewTest) WithIndex() *KrewTest {
+	k.initializeIndex()
+	return k
+}
+
+// WithEnv sets an environment variable for the krew run.
+func (k *KrewTest) WithEnv(key string, value interface{}) *KrewTest {
+	if key == "KREW_ROOT" {
+		glog.V(1).Infoln("Overriding KREW_ROOT in tests is forbidden")
+		return k
+	}
+	k.env = append(k.env, fmt.Sprintf("%s=%v", key, value))
+	return k
+}
+
+// RunOrFail runs the krew command and fails the test if the command returns an error.
+func (k *KrewTest) RunOrFail() {
+	k.t.Helper()
+	if err := k.Run(); err != nil {
+		k.t.Fatal(err)
+	}
+}
+
+// Run runs the krew command.
+func (k *KrewTest) Run() error {
+	k.t.Helper()
+
+	cmd := k.cmd(context.Background())
+	glog.V(1).Infoln(cmd.Args)
+
+	start := time.Now()
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "krew %v", k.args)
+	}
+
+	glog.V(1).Infoln("Ran in", time.Since(start))
+	return nil
+}
+
+// RunOrFailOutput runs the krew command and fails the test if the command
+// returns an error. It only returns the standard output.
+func (k *KrewTest) RunOrFailOutput() []byte {
+	k.t.Helper()
+
+	cmd := k.cmd(context.Background())
+	cmd.Stdout, cmd.Stderr = nil, nil
+	glog.V(1).Infoln(cmd.Args)
+
+	start := time.Now()
+	out, err := cmd.Output()
```

Comment:
you will need stderr for debugging. So perhaps CombinedOutput? 
We don’t have any commands providing a parseable output right now so it’s probably ok.